pkg/assets/assets.go

```diff
@@ -66,3 +76,17 @@ func Generate(fileName string, packageName string, variableName string, dirs map
 		VariableName: variableName,
 	})
 }
+
+// Extract creates a directory named dst including any missing parents and extracts the assets at
+// src into the created directory. An error is returned if the directory already exists.
+//
+// The assets are read either from data embedded in the binary or from the filesystem, depending on
+// whether the LOKOCTL_USE_FS_ASSETS environment variable is specified.
+func Extract(src, dst string) error {
+	walk := CopyingWalker(dst, 0755)
+	if err := Assets.WalkFiles(src, walk); err != nil {
+		return errors.Wrap(err, "failed to walk assets")
+	}
+
+	return nil
+}
```

Comment:
I'd rephrase things to make them more clear.
```suggestion
// Extract extracts given assets path into local file system. All parent directories will be automatically
// created.
//
// The assets are read either from data embedded in the binary or from the filesystem, depending on
// whether the LOKOCTL_USE_FS_ASSETS environment variable is specified.
func Extract(assetsPath, fsPath string) error {
	walk := CopyingWalker(fsPath, 0755)
	if err := Assets.WalkFiles(assetsPath, walk); err != nil {
		return fmt.Errorf("failed to walk assets: %w", err)
	}

	return nil
}
```

Reply:
I've edited the comment to make it more explicit. I've also added two unit tests as I had to verify the function's behavior. It turns out it doesn't return an error if the target directory exists although the old comment said so.